staging/src/k8s.io/cri-api/pkg/apis/runtime/v1/api.proto

```diff
@@ -1679,6 +1679,8 @@ message MemoryUsage {
     UInt64Value page_faults = 6;
     // Cumulative number of major page faults.
     UInt64Value major_page_faults = 7;
+    // Total commit memory in use. Commit memory is total of physical and virtual memory in use.
+    UInt64Value commit_memory_bytes = 8;
```

Comment:
Do we actually envision using this for Linux? Trying to grok why it was added here as well (unless we mix and match using MemoryUsage and WindowsMemoryUsage for different scenarios

Reply:
We weren't sure if this is something folks would want to see in Linux.  There are fields in MemoryUsage that are Linux only.   

Two reasons we thought we wanted to start the conversation and why we thought it might be ok to add here:

- if we only add it to the Windows Memory field, then it will not be usable until the CRI only KEP comes out of Alpha and is used in various upstream projects.  This is causing enough issues for workloads today that it seems like exposing it would be good idea.

- This behaves like a bug.  The Limit set on the job object in Windows uses commit memory as its upper bound.  This means there are edge cases that cause a pod to crash before scaling up with an HPA due to the current use of WorkingSetBytes for Windows.

Reply:
pss?

Reply:
@mikebrow sorry I don't follow...

Reply:
Was a question to @haircommander if pss (proportional set size) in linux is the closest thing we can get to commit memory in windows.

Reply:
> if we only add it to the Windows Memory field, then it will not be usable until the CRI only KEP comes out of Alpha and is used in various upstream projects

then we should direct attention to getting CRI stats out of alpha :D 

> This behaves like a bug. The Limit set on the job object in Windows uses commit memory as its upper bound. This means there are edge cases that cause a pod to crash before scaling up with an HPA due to the current use of WorkingSetBytes for Windows.

it's not clear to me what the ultimate goal of adding this field. will HPA read this field exposed by kubelet to create the upper bound?

> pss?

I suppose, though I think generally we should be wary about adding fields to the kubelet's /stats/summary. If there is a reason for an external entity (HPA on windows, as it seems this case is highlighting), then definitely. but just adding a field to the linux object to match the windows (When we intentionally kept them separate for this) seems needless to me

Reply:
> it's not clear to me what the ultimate goal of adding this field. will HPA read this field exposed by kubelet to create the upper bound?

Yes, we would need to modify the HPA and Metric Server to read from this field in order to scale properly and avoid Out of memory errors.

> I suppose, though I think generally we should be wary about adding fields to the kubelet's /stats/summary. If there is a reason for an external entity (HPA on windows, as it seems this case is highlighting), then definitely. but just adding a field to the linux object to match the windows (When we intentionally kept them separate for this) seems needless to me

I agree we want to be careful about what we add to `/stats/summary`, and glad we did keep them separate moving forward for this reason.  I guess there isn't a field that would make sense to both Windows and Linux?

Reply:
Considering that the `MemoryUsage` is used as Linux only once we move to CRI only stats and there is no field that makes sense for both Windows and Linux, I am thinking right now that it might not make sense to add it here.

Reply:
>  I am thinking right now that it might not make sense to add it here.

+1 I agree

Reply:
> I am thinking right now that it might not make sense to add it here.

Only adding it to `WindowsMemoryUsage` correct?
If that is the case +1 from me as well

Reply:
> Only adding it to `WindowsMemoryUsage` correct? If that is the case +1 from me as well

Right, this would be my preferred. My original comment was trying to see if there was some reason it was added to both here, but it seems we can get away with just Windows.

Reply:
nod

Reply:
Thanks for the inputs. I updated the PR to reflect the changes.

Apologies for the delay. I was on vacation and just got back.